hiero-dependency-versions/build.gradle.kts

```diff
@@ -97,7 +99,7 @@ dependencies.constraints {
     api("com.goterl:lazysodium-java:5.1.4") { because("lazysodium.java") }
     api("net.i2p.crypto:eddsa:0.3.0") { because("net.i2p.crypto.eddsa") }
     api("org.antlr:antlr4-runtime:4.13.2") { because("org.antlr.antlr4.runtime") }
-    api("commons-codec:commons-codec:1.15") { because("org.apache.commons.codec") }
+    api("commons-codec:commons-codec:1.17.1") { because("org.apache.commons.codec") }
```

Comment:
The version upgrades that are unrelated to this PR are also done through https://github.com/hashgraph/hedera-services/pull/17232.

Once that is merged (and this PR is rebased) the changes will disappear here.